### PR TITLE
fix(badge): investigate failing RTL test

### PR DESCRIPTION
**Pull request by mihai-albu-sage**
### Proposed behaviour
Testing CSS changes based on pseudo-selectors like `:focus` and `:hover` can be tricky because Jest itself doesn't interpret CSS or pseudo-classes natively. CSS-influenced behavior, like changes to styles based on `:focus` requires the browser's rendering engine, which Jest doesn't simulate.

The `:focus` and `:hover` pseudo selectors have been emulated using React Synthetic events (`onFocus`, `onBlur`, `onMouseEnter`, `onMouseLeave`) and the CSS changes are now applied based on two props (`isFocused` and `isHovered`) that are now passed to the `StyledBadge` component.

### Current behaviour
When using `toBeVisible()` the test is failing due to visibility being changed by pseudo-selectors.

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

**Review by Parsium — DISMISSED** [review]
Nice one for fixing this @mihai-albu-sage! 👏🏼

**Review by Parsium — COMMENTED** [review]
Great job investigating and fixing this @mihai-albu-sage! 👏🏼

File: src/components/badge/badge.test.tsx [hunk 1]
```diff
@@ -32,9 +32,27 @@ describe("Badge", () => {
...
+
+    badgeButton.blur();
+
+    expect(badgeText).toBeVisible();
+
+    fireEvent.mouseEnter(badgeButton);
+
+    expect(badgeText).not.toBeVisible();
+
+    fireEvent.mouseLeave(badgeButton);
```
[the remark this answers is not in the document]
Comment (mihai-albu-sage) [reply src/components/badge/badge.test.tsx]: Thank you for the suggestion. Test have been refactored to split the various use cases.

File: src/components/badge/badge.test.tsx [hunk 2]
```diff
@@ -32,9 +32,27 @@ describe("Badge", () => {
...
+
+    badgeButton.focus();
+
+    expect(badgeText).not.toBeVisible();
+
+    badgeButton.blur();
+
+    expect(badgeText).toBeVisible();
+
+    fireEvent.mouseEnter(badgeButton);
```
[the remark this answers is not in the document]
Comment (mihai-albu-sage) [reply src/components/badge/badge.test.tsx]: Done, refactored with `userEvent`

**Review by Parsium — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Pull request merged**